.github/workflows/ci.yml

```diff
@@ -183,19 +198,25 @@ jobs:
 
     steps:
 
-      - name: Check out code
+      - name: 1. Check out code
         uses: actions/checkout@v3 # https://github.com/actions/checkout
         with:
           fetch-depth: '0' # https://github.com/shipkit/shipkit-changelog#fetch-depth-on-ci
 
-      - name: Set up Java 11
+      - name: 2. Set up Java for running Gradle build
         uses: actions/setup-java@v3
         with:
           distribution: 'zulu'
-          java-version: 11
-
-      - name: Build and release
-        run: ./gradlew githubRelease publishToSonatype closeAndReleaseStagingRepository releaseSummary
+          java-version: 17
```

Comment:
This line means (*).

Before:
 * Gradle Java 11* producing Java 11 bytecode (because targetCompatibility).
 * Test matrix Java 11 producing Java 11 bytecode running on Java 11
 * Test matrix Java 17 producing Java 11 bytecode running on Java 17

After:
 * Gradle Java 17* producing Java 11 bytecode (because targetCompatibility).
 * Test matrix Java 17 producing Java 11 bytecode running on Java 11
 * Test matrix Java 17 producing Java 11 bytecode running on Java 17

The key here is that the built bytecode is always the same, and is tested on different runtimes. Same as user's envs.

I think unless Java 21+ LTS [deprecates target=11](https://inside.java/2022/09/26/quality-heads-up/), we can keep using this pattern. If deprecation happens we can increase minimum, or force a toolchain, but that would remove optimizations/fixes of javac in Java 12-17.